console/frontend/src/main/frontend/src/app/views/security-items/security-items.service.ts

```diff
@@ -81,16 +163,28 @@ export type Links = { links: Link[] };
   providedIn: 'root',
 })
 export class SecurityItemsService {
-  constructor(
-    private http: HttpClient,
-    private appService: AppService,
-  ) {}
+  private securityItemsCache: SecurityItems | null = null;
+  private endpointsWithRolesCache: Links | null = null;
+  private readonly http: HttpClient = inject(HttpClient);
+  private readonly appService: AppService = inject(AppService);
 
   getSecurityItems(): Observable<SecurityItems> {
-    return this.http.get<SecurityItems>(`${this.appService.absoluteApiPath}securityitems`);
+    if (this.securityItemsCache) {
+      return of(this.securityItemsCache);
+    }
+
+    return this.http
+      .get<SecurityItems>(`${this.appService.absoluteApiPath}securityitems`)
+      .pipe(tap((data) => (this.securityItemsCache = data)));
   }
 
   getEndpointsWithRoles(): Observable<Links> {
-    return this.http.get<Links>(`${this.appService.absoluteApiPath}?allowedRoles=true`);
+    if (this.endpointsWithRolesCache) {
+      return of(this.endpointsWithRolesCache);
+    }
```

Comment:
Maybe this should have a way to be cleared or updated too, for example when a user logs out

Reply:
Same.